test/azure_blobs_exception_tests.jl

```diff
@@ -144,3 +144,141 @@
+        http_server = HTTP.serve!(tcp_server) do request::HTTP.Request
+            if request.method == "GET" && request.target == "/$account/$container/_this_file_does_not_exist"
+                # This is the exploratory ping from connect_and_test in lib.rs
+                return HTTP.Response(404, "Yup, still doesn't exist")
+            end
+            nretries[] += 1
+            response = isnothing(headers) ? HTTP.Response(response_status, response_body) : HTTP.Response(response_status, headers, response_body)
+            return response
+        end
+
+        baseurl = "http://127.0.0.1:$port/$account/$container/"
+        creds = AzureCredentials(account, container, shared_key_from_azurite, baseurl)
+
+        try
+            method === :GET && blob_get!(joinpath(baseurl, "blob"), zeros(UInt8, 5), creds)
+            method === :PUT && blob_put(joinpath(baseurl, "blob"), codeunits("a,b,c"), creds)
+            @test false # Should have thrown an error
+        catch e
+            @test e isa ErrorException
+            @test occursin(string(response_status), e.msg)
+            response_status < 500 && (@test occursin("response body from the dummy server", e.msg))
+        finally
+            close(http_server)
+        end
+        wait(http_server)
+        return nretries[]
+    end
+
+    # See https://learn.microsoft.com/en-us/rest/api/searchservice/http-status-codes
+
+    @testset "400: Bad Request" begin
+        # Returned when there's an error in the request URI, headers, or body. The response body
+        # contains an error message explaining what the specific problem is.
+        nretries = test_status(:GET, 400)
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 400)
+        @test nretries == 1 broken=true
+    end
+
+    @testset "403: Forbidden" begin
+        # Returned when you pass an invalid api-key.
+        nretries = test_status(:GET, 403)
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 403)
+        @test nretries == 1 broken=true
+    end
+
+    @testset "404: Not Found" begin
+        nretries = test_status(:GET, 404)
+        @test nretries == 1
+    end
+
+    @testset "405: Method Not Supported" begin
+        nretries = test_status(:GET, 405, ["Allow" => "PUT"])
+        @test nretries == 1 broken=true
+        nretries = test_status(:PUT, 405, ["Allow" => "GET"])
+        @test nretries == 1 broken=true
+    end
+
+    @testset "409: Conflict" begin
+        # Returned when write operations conflict.
+        # NOTE: We currently don't retry but maybe we should? This is probably a case where the
+        # retry logic should add more noise to the backoff so that multiple writers don't collide on retry.
+        nretries = test_status(:GET, 409)
+        @test nretries == max_retries broken=true
+        nretries = test_status(:PUT, 409)
+        @test nretries == max_retries broken=true
```

Comment:
I don't think we should retry these and the behavior after the rust lib fix doesn't retry. 409 looks like it is used for immutable blobs and things like that.